src/main/ml-modules/root/com.marklogic.smart-mastering/algorithms/double-metaphone.xqy

```diff
@@ -57,12 +57,20 @@ declare
         spell:suggest($dictionary, $value, $spell-options)[fn:not(fn:lower-case(.) = fn:lower-case($value))]
     where fn:exists($expanded-values)
     return
-      cts:element-value-query(
-        $qname,
-        $expanded-values,
-        "case-insensitive",
-        $expand-xml/@weight
-      )
+      if ($options-xml/match:data-format = $const:FORMAT-JSON) then
+        cts:json-property-value-query(
+          fn:string($qname),
+          $expanded-values,
+          "case-insensitive",
+          $expand-xml/@weight
+        )
+      else
```

Comment:
Something I just started doing in my last PR is to explicitly check that the value is either `FORMAT-JSON` or `FORMAT-XML` and throwing an exception if not. Since we're talking about user-entered content, we should be prepared for it to be wrong. So, if data-format = FORMAT-JSON then do JSON stuff; else if data-format = FORMAT-XML then do XML stuff; else throw SM-INVALID-FORMAT. I think that's worth it to expose errors.